src/lib.rs

```diff
@@ -49,6 +52,16 @@ impl FileError {
                     ret.push_str(&epath.display().to_string());
                     ret.push_str(" does not exist!");
                 }
+                Anchor(url, anchor) => ret.push_str(&format!(
+                    "\n\tAnchor #{} in linked file at URL {} does not exist!",
+                    anchor, url
+                )),
+                AnchorRange(url, anchor, missing) => ret.push_str(&format!(
+                    "\n\tRanged anchor #{} in linked file at path {} is missing parts: {}",
+                    anchor,
+                    url,
+                    missing.join(", ")
+                )),
```

Comment:
This is duplicated in `HttpError`. Rather than have it in both places, can you add a new `AnchorError` struct with this `Display` impl, then have an `Anchor` variant on both `FileError` and `HttpError` that delegates to it?

Reply:
I found myself refactoring this a lot, I didn't do this exactly but decided that for CheckErrors there should be four different kinds: file, http, fragment (renamed from anchor) and io errors. Do let me know what you think, I'm not entirely certain about this.